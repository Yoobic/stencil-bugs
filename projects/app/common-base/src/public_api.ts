export { AppCommonBaseModule } from './lib/app-common-base.module';

export * from './lib/bases/app-chat-page/app-chat-page.component';
export * from './lib/bases/app-entity-action-page/app-entity-action-page.component';
export * from './lib/bases/app-form-dynamic-page/app-form-dynamic-page.component';
export * from './lib/bases/app-grid-dialog-page/app-grid-dialog-page.component';
export * from './lib/bases/app-menu-page/app-menu-page.component';
export * from './lib/bases/app-root/app-root.component';
export * from './lib/bases/app-search-page/app-search-page.component';
export * from './lib/bases/app-walkthrough-page/app-walkthrough-page.component';
export * from './lib/bases/app-whatsnew-page/app-whatsnew-page.component';
export * from './lib/bases/base-page/base-page.component';

export * from './lib/components/app-profile/app-profile.component';
export * from './lib/components/app-login/app-login.component';
export * from './lib/components/app-grid/app-grid.component';
export * from './lib/components/app-form-dynamic/app-form-dynamic.component';

export * from './lib/services/dialog/dialog.service';
export * from './lib/services/utils/utils.service';

export * from './lib/animations/animations';
export * from './lib/animations/transitions';

export * from './lib/modals/app-result-dialog-page/app-result-dialog-page.component';