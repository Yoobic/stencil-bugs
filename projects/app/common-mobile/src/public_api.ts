export { AppCommonMobileModule } from './lib/app-common-mobile.module';

export * from './lib/components/app-root/app-root.component';

export * from './lib/modals/app-grid-dialog-page/app-grid-dialog-page.component';
export * from './lib/modals/app-whatsnew-page/app-whatsnew-page.component';
export * from './lib/modals/app-termsandconditions-page/app-termsandconditions-page.component';
export * from './lib/modals/app-resetpassword-page/app-resetpassword-page.component';
export * from './lib/modals/app-walkthrough-page/app-walkthrough-page.component';
export * from './lib/modals/app-chat-page/app-chat-page.component';
export * from './lib/modals/app-entity-action-page/app-entity-action-page.component';
export * from './lib/modals/app-search-page/app-search-page.component';
export * from './lib/modals/app-form-dynamic-page/app-form-dynamic-page.component';