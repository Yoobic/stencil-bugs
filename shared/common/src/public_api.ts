export { CommonModule } from './lib/common.module';

export * from './lib/decorators/mixin/mixin.decorator';
export * from './lib/decorators/persistent/persistent.decorator';
export * from './lib/directives/long-press/long-press.directive';

export * from './lib/interfaces/animate-state/animate-state.class';
export * from './lib/interfaces/animations/animations.class';
export * from './lib/interfaces/colors/colors.interface';
export * from './lib/interfaces/config-constants/config-constants.interface';
export * from './lib/interfaces/position/position.interface';

export * from './lib/pipes/array-from-number/array-from-number.pipe';
export * from './lib/pipes/cloudinary/cloudinary.pipe';
export * from './lib/pipes/date-format/date-format.pipe';
export * from './lib/pipes/file-size/file-size.pipe';
export * from './lib/pipes/highlight/highlight.pipe';
export * from './lib/pipes/https/https.pipe';
export * from './lib/pipes/keys/keys.pipe';
export * from './lib/pipes/limit-to/limit-to.pipe';
export * from './lib/pipes/map-values/map-values.pipe';
export * from './lib/pipes/newline/newline.pipe';
export * from './lib/pipes/nohtml/nohtml.pipe';
export * from './lib/pipes/order-by/order-by.pipe';
export * from './lib/pipes/round/round.pipe';
export * from './lib/pipes/safe/safe.pipe';
export * from './lib/pipes/take/take.pipe';
export * from './lib/pipes/time-ago/time-ago.pipe';
export * from './lib/pipes/timer/timer.pipe';

export * from './lib/services/config-constants/config-constants.service';
export * from './lib/services/core-config/core-config.service';
export * from './lib/services/dom-adapter/dom-adapter.service';
export * from './lib/services/exif-restorer/exif-restorer.service';
export * from './lib/services/geo-location/geo-location.service';
export * from './lib/services/local-forage/local-forage.service';
export * from './lib/services/local-storage/local-storage.service';
export * from './lib/services/log/log.service';
export * from './lib/services/network/network.service';
export * from './lib/services/promise/promise.service';
export * from './lib/services/resize/resize.service';
export * from './lib/services/ruler/ruler.service';
export * from './lib/services/utils/utils.service';

export * from './lib/mixins/lodash-mixins/lodash-mixins';