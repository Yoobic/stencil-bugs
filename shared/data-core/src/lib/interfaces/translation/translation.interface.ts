import { FormFieldType, ITranslation } from '@shared/interfaces';
import { Model } from '../../decorators/model/model.decorator';
import { Editable } from '../../decorators/editable/editable.decorator';
import { Searchable } from '../../decorators/searchable/searchable.decorator';
import { Translate } from '@shared/translate';
import { Googletranslate } from '../../services/googletranslate/googletranslate.service';
import { Requestor } from '../../services/requestor/requestor.service';
import { getGroupsTransform } from '../condition/condition.interface';

import { capitalize } from 'lodash-es';

export function translateButtonHandler(buttonValue, fieldControls, data, form, rq: Requestor) {
    let language = fieldControls.language.value;
    let value = fieldControls.value.value;
    if (value && language) {
        fieldControls[language].markAsTouched();
        fieldControls[language].setValue(value);
        Googletranslate.getAll(value, language, rq).subscribe((values: Array<{ language: string; value: string; }>) => {
            let isCapitalized = capitalize(value) === value;
            values.forEach(v => {
                if (fieldControls && fieldControls[v.language] && !fieldControls[v.language].value) {
                    fieldControls[v.language].markAsTouched();
                    fieldControls[v.language].setValue(isCapitalized ? capitalize(v.value) : v.value);
                }
            });
        });
    }
}

export function resetButtonHandler(buttonValue, fieldControls, data) {
    Translate.availablesLanguage.forEach(l => {
        if (fieldControls[l] && fieldControls[l].setValue) {
            fieldControls[l].setValue(null);
        }
    });
}

@Model({
    className: 'Translation',
    collectionName: 'translations',
    fields: ['*'],
    include: []
})
export class Translation extends ITranslation {
    @Editable('Translation', { title: 'ID', visible: false, forceExport: true })
    _id?: string;

    @Editable('Translation', { type: FormFieldType.autocomplete, title: 'LANGUAGE', translate: true, values: Translate.availablesLanguage, clearable: false, required: true, autoselect: true, columnDefinition: { width: 40 } })
    language: string;

    @Editable('Translation', { required: true, type: FormFieldType.text })
    @Searchable('Translation') key: string;

    @Editable('Translation', {
        required: true,
        type: FormFieldType.text
    })
    value: string;

    @Editable('Translation', {
        type: FormFieldType.button,
        title: 'TRANSLATE',
        flex: 50,
        suppressExport: true,
        handler: translateButtonHandler
    })
    translateButton?: any;

    @Editable('Translation', {
        type: FormFieldType.button,
        title: 'RESET',
        suppressExport: true,
        color: 'assertive light',
        handler: resetButtonHandler,
        flex: 50
    })
    resetButton?: any;

    @Editable('Translation', { required: true, type: FormFieldType.text }) en: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) us: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) fr: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) es: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) pl: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) nl: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) de: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) it: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) ru: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) zhs: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) zht: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) pt: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) kr: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) ja: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) ua: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) he: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) ar: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) cz: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) th: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) tr: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) bg: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) el: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) sl: string;

    @Editable('Translation', { required: true, type: FormFieldType.text }) sk: string;

    @Editable('Translation', { type: FormFieldType.toggle, title: 'UNVALIDATEREASON' })
    isReject: boolean;

    @Editable('Translation', { title: 'GROUPS', type: FormFieldType.autocomplete, collectionName: 'groups', filters: [[{ field: 'isRole', operator: { _id: 'neq' }, value: true }, { field: 'type', operator: { _id: 'nin' }, value: ['plan'] }]], hiddenFields: ['isRole', 'type'], idOnly: true, mapTransform: getGroupsTransform, multiple: true, clearable: false, required: true })
    group: string | Array<string>;

    @Editable('Translation', { type: FormFieldType.toggle, title: 'ANNOTATE' })
    isPhotoAnnotation: boolean;
}
