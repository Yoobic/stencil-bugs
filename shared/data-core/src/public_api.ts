export { DataCoreModule } from './lib/data-core.module';

export * from './lib/decorators/editable/editable.decorator';
export * from './lib/decorators/mapping/mapping.decorator';
export * from './lib/decorators/model/model.decorator';
export * from './lib/decorators/searchable/searchable.decorator';

export * from './lib/interfaces/aggregate-log/aggregate-log.interface';
export * from './lib/interfaces/algorithm/algorithm.interface';
export * from './lib/interfaces/autorenew-config/autorenew-config.interface';
export * from './lib/interfaces/backup/backup.interface';
export * from './lib/interfaces/catalog/catalog.interface';
export * from './lib/interfaces/chart-definition/chart-definition.interface';
export * from './lib/interfaces/condition/condition.interface';
export * from './lib/interfaces/condition/icondition.interface';
export * from './lib/interfaces/constants/constants.interface';
export * from './lib/interfaces/currency/currency.interface';
export * from './lib/interfaces/dashboard/dashboard.interface';
export * from './lib/interfaces/entity/entity.interface';
export * from './lib/interfaces/field-selector/field-selector.interface';
export * from './lib/interfaces/file/file.interface';
export * from './lib/interfaces/folder/folder.interface';
export * from './lib/interfaces/group/group.interface';
export * from './lib/interfaces/kpi/kpi.interface';
export * from './lib/interfaces/location-type/location-type.interface';
export * from './lib/interfaces/location/location.interface';
export * from './lib/interfaces/mapping/mapping.interface';
export * from './lib/interfaces/mission-description/mission-description.interface';
export * from './lib/interfaces/model/model.interface';
export * from './lib/interfaces/notification-event/notification-event.interface';
export * from './lib/interfaces/notification/notification.interface';
export * from './lib/interfaces/operation/operation.interface';
export * from './lib/interfaces/product/product.interface';
export * from './lib/interfaces/response-object/response-object.interface';
export * from './lib/interfaces/slide/slide.interface';
export * from './lib/interfaces/sort/sort.interface';
export * from './lib/interfaces/tableau/tableau.interface';
export * from './lib/interfaces/tag-group/tag-group.interface';
export * from './lib/interfaces/tag/tag.interface';
export * from './lib/interfaces/tenant/tenant.interface';
export * from './lib/interfaces/todo/todo.interface';
export * from './lib/interfaces/translation/translation.interface';
export * from './lib/interfaces/user/user.interface';
export * from './lib/interfaces/workplace/workplace.interface';

export * from './lib/pipes/authorized/authorized.pipe';
export * from './lib/pipes/currency/currency.pipe';
export * from './lib/pipes/filter/filter.pipe';
export * from './lib/pipes/username/username.pipe';
export * from './lib/pipes/useremail/useremail.pipe';
export * from './lib/pipes/userinitial/userinitial.pipe';

export * from './lib/services/activity/activity.service';
export * from './lib/services/algorithms/algorithms.service';
export * from './lib/services/authentication/authentication.guard';
export * from './lib/services/authentication/authentication.service';
export * from './lib/services/authentication/current-session.resolver';
export * from './lib/services/backup/backup.service';
export * from './lib/services/box/box.service';
export * from './lib/services/blog/blog.service';
export * from './lib/services/links/links.service';
export * from './lib/services/broker/broker.service';
export * from './lib/services/config/config.service';
export { Dashboard as DashboardCore } from './lib/services/dashboard/dashboard.service';
export * from './lib/services/data-loader/data-loader.service';
export * from './lib/services/files/files.service';
export * from './lib/services/files/files-broker.service';
export * from './lib/services/googlemaps/googlemaps.service';
export * from './lib/services/googletranslate/googletranslate.service';
export * from './lib/services/http-interceptor/http-interceptor.service';
export * from './lib/services/loading-bar/loading-bar.service';
export * from './lib/services/locations/locations.service';
export * from './lib/services/loopback/loopback.service';
export * from './lib/services/models/models.service';
export * from './lib/services/print/print.service';
export * from './lib/services/push/push.service';
export * from './lib/services/requestor/requestor.service';
export * from './lib/services/security/security.service';
export * from './lib/services/session/session.service';
export * from './lib/services/smartloc/smartloc.service';
export * from './lib/services/users/users.service';
export * from './lib/services/version/version.service';
export * from './lib/services/workplace/workplace.service';
export * from './lib/services/xlsx/xlsx.service';

export * from './lib/validators/password-validator/password-validator';
export * from './lib/validators/username-validator/username-validator';
