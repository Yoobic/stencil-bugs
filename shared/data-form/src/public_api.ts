
export { DataFormModule } from './lib/data-form.module';

export * from './lib/interfaces/custom-form-field/custom-form-field.interface';
export * from './lib/interfaces/form-field/form-field.class';
export * from './lib/interfaces/gridstate/gridstate.interface';
export * from './lib/interfaces/cache/cache.interface';
export * from './lib/interfaces/custom-model/custom-model.interface';

export * from './lib/services/cell-renderer/cell-renderer.service';
export * from './lib/services/chart/chart.service';
export * from './lib/services/form-dynamic-builder/form-dynamic-builder.service';
export * from './lib/services/conditions/conditions.service';
export * from './lib/services/mapping/mapping.service';
export * from './lib/services/custom-models/custom-models.service';

export * from './lib/validators/form-filter-field-validator/form-filter-field-validator';
export * from './lib/validators/form-input-base-email-validator/form-input-base-email-validator';
export * from './lib/validators/form-input-base-number-validator/form-input-base-number-validator';
export * from './lib/validators/form-input-between-validator/form-input-between-validator';
export * from './lib/validators/form-input-checklist-validator/form-input-checklist-validator';
export * from './lib/validators/form-input-missionfield-validator/form-input-missionfield-validator';
export * from './lib/validators/form-input-multiphotos-validator/form-input-multiphotos-validator';
export * from './lib/validators/form-input-photo-validator/form-input-photo-validator';
export * from './lib/validators/form-input-select-validator/form-input-select-validator';
export * from './lib/validators/password-are-not-matching-validator/password-are-not-matching-validator';

export * from './lib/renderers/address/address-renderer.component';
export * from './lib/renderers/boolean/boolean-renderer.component';
export * from './lib/renderers/button/button-renderer.component';
export * from './lib/renderers/checklist/checklist-renderer.component';
export * from './lib/renderers/conformity-progress/conformity-progress-renderer.component';
export * from './lib/renderers/conformity-relative-progress/conformity-relative-progress-renderer.component';
export * from './lib/renderers/custom-model/custom-model-renderer.component';
export * from './lib/renderers/datetime/datetime-renderer.component';
export * from './lib/renderers/default/default-renderer.component';
export * from './lib/renderers/delta/delta-renderer.component';
export * from './lib/renderers/distance/distance-renderer.component';
export * from './lib/renderers/duration/duration-renderer.component';
export * from './lib/renderers/location/location-renderer.component';
export * from './lib/renderers/mission-status/mission-status-renderer.component';
export * from './lib/renderers/mission-validation/mission-validation-renderer.component';
export * from './lib/renderers/number/number-renderer.component';
export * from './lib/renderers/multiphotos/multiphotos-renderer.component';
export * from './lib/renderers/percentage/percentage-renderer.component';
export * from './lib/renderers/photo/photo-renderer.component';
export * from './lib/renderers/progress/progress-renderer.component';
export * from './lib/renderers/time/time-renderer.component';
export * from './lib/renderers/timer/timer-renderer.component';
export * from './lib/renderers/todo/todo-renderer.component';
export * from './lib/renderers/user-full-name/user-full-name-renderer.component';
export * from './lib/renderers/validation-progress/validation-progress-renderer.component';
export * from './lib/renderers/video/video-renderer.component';