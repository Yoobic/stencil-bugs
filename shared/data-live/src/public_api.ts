
export { DataLiveModule } from './lib/data-live.module';

export { Channel as ChannelInterface } from './lib/interfaces/channel/channel.interface';
export * from './lib/interfaces/channels/channels.interface';
export * from './lib/interfaces/message/message.interface';
export * from './lib/interfaces/message/message.model';

export * from './lib/services/channel/channel.service';
export * from './lib/services/intercom/intercom.service';
export * from './lib/services/messages/messages.service';
export * from './lib/services/pubnub/pubnub.service';
export * from './lib/services/track/track.service';
export * from './lib/services/twilio-token/twilio-token.service';
export * from './lib/services/raven-error-handler/raven-error-handler.service';