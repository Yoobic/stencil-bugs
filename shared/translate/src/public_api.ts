export { TranslateModule } from './lib/translate.module';

export * from './lib/pipes/distance/distance.pipe';
export * from './lib/pipes/filter/filter.pipe';
export * from './lib/pipes/polyglot/polyglot.pipe';

export * from './lib/services/translate/translate.service';