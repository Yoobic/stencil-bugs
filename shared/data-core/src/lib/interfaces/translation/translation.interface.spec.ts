import { Translation } from './translation.interface';
import { Models } from '../../../public_api';

describe('shared: data-core: interfaces: Translation', () => {
    it('should create an entity', () => {
        let entity = new Translation();
        expect(entity._id).toBeUndefined();
        let model = Models.getModel('Translation');
        expect(model).toBeTruthy();
        expect(model).toMatchSnapshot();
    });
});