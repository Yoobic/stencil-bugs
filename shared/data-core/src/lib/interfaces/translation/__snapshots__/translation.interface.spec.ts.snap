// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared: data-core: interfaces: Translation should create an entity 1`] = `
IModel {
  "_formFields": Array [
    Object {
      "filterName": "_id",
      "forceExport": true,
      "name": "_id",
      "title": "ID",
      "visible": false,
    },
    Object {
      "autoselect": true,
      "clearable": false,
      "columnDefinition": Object {
        "width": 40,
      },
      "filterName": "language",
      "name": "language",
      "required": true,
      "title": "LANGUAGE",
      "translate": true,
      "type": "autocomplete",
      "values": Array [
        "en",
        "us",
        "fr",
        "de",
        "es",
        "nl",
        "pl",
        "it",
        "ru",
        "zhs",
        "zht",
        "pt",
        "kr",
        "ja",
        "ua",
        "he",
        "ar",
        "cz",
        "th",
        "tr",
        "bg",
        "el",
        "sl",
        "sk",
      ],
    },
    Object {
      "filterName": "key",
      "name": "key",
      "required": true,
      "title": "KEY",
      "type": "text",
    },
    Object {
      "filterName": "value",
      "name": "value",
      "required": true,
      "title": "VALUE",
      "type": "text",
    },
    Object {
      "filterName": "translateButton",
      "flex": 50,
      "handler": [Function],
      "name": "translateButton",
      "suppressExport": true,
      "title": "TRANSLATE",
      "type": "button",
    },
    Object {
      "color": "assertive light",
      "filterName": "resetButton",
      "flex": 50,
      "handler": [Function],
      "name": "resetButton",
      "suppressExport": true,
      "title": "RESET",
      "type": "button",
    },
    Object {
      "filterName": "en",
      "name": "en",
      "required": true,
      "title": "EN",
      "type": "text",
    },
    Object {
      "filterName": "us",
      "name": "us",
      "required": true,
      "title": "US",
      "type": "text",
    },
    Object {
      "filterName": "fr",
      "name": "fr",
      "required": true,
      "title": "FR",
      "type": "text",
    },
    Object {
      "filterName": "es",
      "name": "es",
      "required": true,
      "title": "ES",
      "type": "text",
    },
    Object {
      "filterName": "pl",
      "name": "pl",
      "required": true,
      "title": "PL",
      "type": "text",
    },
    Object {
      "filterName": "nl",
      "name": "nl",
      "required": true,
      "title": "NL",
      "type": "text",
    },
    Object {
      "filterName": "de",
      "name": "de",
      "required": true,
      "title": "DE",
      "type": "text",
    },
    Object {
      "filterName": "it",
      "name": "it",
      "required": true,
      "title": "IT",
      "type": "text",
    },
    Object {
      "filterName": "ru",
      "name": "ru",
      "required": true,
      "title": "RU",
      "type": "text",
    },
    Object {
      "filterName": "zhs",
      "name": "zhs",
      "required": true,
      "title": "ZHS",
      "type": "text",
    },
    Object {
      "filterName": "zht",
      "name": "zht",
      "required": true,
      "title": "ZHT",
      "type": "text",
    },
    Object {
      "filterName": "pt",
      "name": "pt",
      "required": true,
      "title": "PT",
      "type": "text",
    },
    Object {
      "filterName": "kr",
      "name": "kr",
      "required": true,
      "title": "KR",
      "type": "text",
    },
    Object {
      "filterName": "ja",
      "name": "ja",
      "required": true,
      "title": "JA",
      "type": "text",
    },
    Object {
      "filterName": "ua",
      "name": "ua",
      "required": true,
      "title": "UA",
      "type": "text",
    },
    Object {
      "filterName": "he",
      "name": "he",
      "required": true,
      "title": "HE",
      "type": "text",
    },
    Object {
      "filterName": "ar",
      "name": "ar",
      "required": true,
      "title": "AR",
      "type": "text",
    },
    Object {
      "filterName": "cz",
      "name": "cz",
      "required": true,
      "title": "CZ",
      "type": "text",
    },
    Object {
      "filterName": "th",
      "name": "th",
      "required": true,
      "title": "TH",
      "type": "text",
    },
    Object {
      "filterName": "tr",
      "name": "tr",
      "required": true,
      "title": "TR",
      "type": "text",
    },
    Object {
      "filterName": "bg",
      "name": "bg",
      "required": true,
      "title": "BG",
      "type": "text",
    },
    Object {
      "filterName": "el",
      "name": "el",
      "required": true,
      "title": "EL",
      "type": "text",
    },
    Object {
      "filterName": "sl",
      "name": "sl",
      "required": true,
      "title": "SL",
      "type": "text",
    },
    Object {
      "filterName": "sk",
      "name": "sk",
      "required": true,
      "title": "SK",
      "type": "text",
    },
    Object {
      "filterName": "isReject",
      "name": "isReject",
      "title": "UNVALIDATEREASON",
      "type": "toggle",
    },
    Object {
      "clearable": false,
      "collectionName": "groups",
      "filterName": "group",
      "filters": Array [
        Array [
          Object {
            "field": "isRole",
            "operator": Object {
              "_id": "neq",
            },
            "value": true,
          },
          Object {
            "field": "type",
            "operator": Object {
              "_id": "nin",
            },
            "value": Array [
              "plan",
            ],
          },
        ],
      ],
      "hiddenFields": Array [
        "isRole",
        "type",
      ],
      "idOnly": true,
      "mapTransform": [Function],
      "multiple": true,
      "name": "group",
      "required": true,
      "title": "GROUPS",
      "type": "autocomplete",
    },
    Object {
      "filterName": "isPhotoAnnotation",
      "name": "isPhotoAnnotation",
      "title": "ANNOTATE",
      "type": "toggle",
    },
  ],
  "appareance": Map {},
  "className": "Translation",
  "collectionName": "translations",
  "feathersService": undefined,
  "fields": Array [
    "*",
  ],
  "include": Array [],
  "isCustom": undefined,
  "mappingFields": Map {},
  "searchSubquery": undefined,
  "searchableFields": Array [
    "key",
  ],
  "type": [Function],
}
`;
